de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/handlers/FileImportHandler.java

```diff
@@ -1,93 +1,71 @@
 import de.tu_bs.cs.isf.e4cf.core.file_structure.util.FileHandlingUtility;
-import de.tu_bs.cs.isf.e4cf.core.util.RCPMessageProvider;
-import de.tu_bs.cs.isf.e4cf.core.util.services.RCPDialogService;
-import de.tu_bs.cs.isf.e4cf.core.util.services.RCPImageService;
 import de.tu_bs.cs.isf.e4cf.core.util.services.RCPSelectionService;
-import de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards.FileImportWizard;
-
+import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
 
 /**
- * A handler that copies files into a selected directory.
+ * A handler that opens the systems file import dialog.
  * 
- * @author Oliver Urbaniak
  */
 public class FileImportHandler {
-	private static final String WINDOW_TITLE = "Import Files";
-	private static final Point WINDOW_SIZE = new Point(800, 600);
-	private static final String FILE_ICON_PATH = "icons/Explorer_View/items/file32.png";
-	
+
 	@Execute
-	public void execute(RCPImageService imageService, RCPDialogService dialogService, RCPSelectionService selectionService,
-			@Named(IServiceConstants.ACTIVE_SHELL) Shell shell, WorkspaceFileSystem fileSystem) {
-		try {
-			Path target = getTargetPath(selectionService);
-			FileImportWizard wizard = buildFileImportWizard(shell, imageService);
-			dialogService.constructDialog(WINDOW_TITLE, WINDOW_SIZE, wizard).open();
-			copyFiles(target, wizard.getSourceFiles(), fileSystem);
-		} catch (IOException e) {
-			RCPMessageProvider.errorMessage(WINDOW_TITLE, e.getMessage());
-		} catch (NullPointerException e ) {
-			RCPMessageProvider.errorMessage(WINDOW_TITLE, e.getMessage());
+	public void execute(RCPSelectionService selectionService, WorkspaceFileSystem workspaceFileSystem) {
+		Path target = Paths.get(workspaceFileSystem.getWorkspaceDirectory().getAbsolutePath());
+
+		final FileChooser fileChooser = new FileChooser();
+		fileChooser.getExtensionFilters().addAll(new ExtensionFilter("All Files", "*.*"));
+		List<File> selectedFiles = fileChooser.showOpenMultipleDialog(null);
+
+		if (selectedFiles != null && !selectedFiles.isEmpty()) {
+			try {
+				target = getTargetPath(selectionService);
+				for (File file : selectedFiles) {
+					workspaceFileSystem.copy(file.toPath(), target);
+				}
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
 		}
-		
 	}
 
-	private void copyFiles(Path target, List<Path> selectedFiles, WorkspaceFileSystem fileSystem) throws IOException {
-		for (Path selectedFile : selectedFiles) {
-			fileSystem.copy(selectedFile, target);
+	@CanExecute
+	public boolean canExecute(RCPSelectionService selectionService) {
+		try {
+			FileTreeElement element = selectionService.getCurrentSelectionFromExplorer();
+			checkForValidSelection(element);
+			return true;
+		} catch (Exception e) {
+			return false;
 		}
 	}
 
-	private Path getTargetPath(RCPSelectionService selectionService) throws NotDirectoryException {
+	private final Path getTargetPath(RCPSelectionService selectionService) throws NotDirectoryException {
```

Comment:
final access modifier does not do anything on private methods

Reply:
removed